build-system/tasks/sweep-experiments/index.js

```diff
@@ -0,0 +1,445 @@
+ */
+const filesContainingPattern = (glob, string) =>
+  getStdoutLines(
+    `grep -El "${cmdEscape(string)}" {${fastGlob.sync(glob).join(',')}}`
+  );
+
+/**
+ * @param {string} fromHash
+ * @return {!Array<string>}
+ */
+const getModifiedSourceFiles = (fromHash) =>
+  getStdoutLines(`git diff --name-only ${fromHash}..HEAD | grep .js`).filter(
+    (file) => !globalWritablePaths.includes(file)
+  );
+
+/**
+ * Runs a jscodeshift transform under this directory.
+ * @param {string} transform
+ * @param {Array<string>=} args
+ * @return {string}
+ */
+const jscodeshift = (transform, args = []) =>
+  getStdout(
+    'npx jscodeshift ' +
+      ` -t ${__dirname}/jscodeshift/${transform} ` +
+      args.join(' ')
+  );
+
+/**
+ * @param {string} id
+ * @return {Array<string>} modified files
+ */
+function removeFromExperimentsConfig(id) {
+  jscodeshift('remove-experiment-config.js', [
+    `--experimentId=${id}`,
+    experimentsConfigPath,
+  ]);
+  return [experimentsConfigPath];
+}
+
+/**
+ * @param {!Object<string, *>} config
+ * @param {string} path
+ * @param {string} id
+ * @return {Array<string>} modified files
+ */
+function removeFromJsonConfig(config, path, id) {
+  delete config[id];
+  writeFileSync(path, JSON.stringify(config, null, 2) + '\n');
+  return [path];
+}
+
+/**
+ * @param {string} id
+ * @param {number} percentage
+ * @return {Array<string>} modified files
+ */
+function removeFromRuntimeSource(id, percentage) {
+  const possiblyModifiedSourceFiles = filesContainingPattern(
+    containRuntimeSource.map((dir) => `${dir}/**/*.js`),
+    id
+  );
+  if (possiblyModifiedSourceFiles.length > 0) {
+    jscodeshift('remove-experiment-runtime.js', [
+      `--isExperimentOnLaunched=${percentage}`,
+      `--isExperimentOnExperiment=${id}`,
+      ...possiblyModifiedSourceFiles,
+    ]);
+  }
+  return possiblyModifiedSourceFiles;
+}
+
+/**
+ * @param {string} id
+ * @param {*} workItem
+ * @param {!Array<string>} modified
+ * @return {? string}
+ */
+function gitCommitSingleExperiment(id, workItem, modified) {
+  const messageParagraphs = [readableRemovalId(id, workItem)];
+  if (workItem.previousHistory.length > 0) {
+    messageParagraphs.push(
+      `Previous history on ${prodConfigPath.split('/').pop()}:`,
+      workItem.previousHistory
+        .map(
+          ({hash, authorDate, subject}) =>
+            `- ${hash} - ${authorDate} - ${subject}`
+        )
+        .join('\n')
+    );
+  }
+  return getStdout(
+    `git add ${modified.join(' ')} && ` +
+      `git commit -m "${cmdEscape(messageParagraphs.join('\n\n'))}"`
+  );
+}
+
+function readableRemovalId(id, {percentage, previousHistory}) {
+  const lastCommit = previousHistory[0];
+  const prefix = lastCommit
+    ? `(${truncateYyyyMmDd(lastCommit.authorDate)}, ${lastCommit.hash})`
+    : 'Remove';
+  return `${prefix} \`${id}\`: ${percentage}`;
+}
+
+/**
+ * @param {number=} daysAgo
+ * @return {!Date}
+ */
+function daysAgo(daysAgo = 365) {
```

Comment:
Intentional to use the same name for the param and the function? (Maybe change one for readability?)